src/odemis/util/transform.py

```diff
@@ -818,3 +818,176 @@ def inverse(self):
+        Scale factors.
+    shear : float
+        Shear factor.
+    coeffs : tuple
+        Tuple of transform coefficients.
+    nlcoeffs : tuple
+        Tuple of non-linear transform coefficients.
+    translation : (2,) array
+        Translation vector.
+
+    References
+    ----------
+    .. [1] J. Stopka, "Polynomial fit for Multi-beam distortions", internal
+    note, 2019
+
+    """
+
+    def __init__(self, coeffs=None, rotation=None, scale=None, shear=None,
+                 nlcoeffs=None, translation=None):
+
+        params = any(param is not None
+                     for param in (rotation, scale, shear, nlcoeffs, translation))
+
+        if params and coeffs is not None:
+            raise ValueError("You cannot specify the transformation "
+                             "coefficients and the implicit parameters at the "
+                             "same time.")
+        elif coeffs is not None:
+            self.coeffs = coeffs
+        else:
+            self.rotation = 0. if rotation is None else rotation
+            self.scale = (1., 1.) if scale is None else scale
+            self.shear = 0. if shear is None else shear
+            self.nlcoeffs = (0., 0., 0., 0., 0., 0., 0.) if nlcoeffs is None else nlcoeffs
+            self.translation = (0., 0.) if translation is None else translation
+
+    @staticmethod
+    def _vandermonde(w):
+        """Return the Vandermonde matrix."""
+        # suffix cc denotes complex conjugate
+        wcc = numpy.conj(w)  # w̄
+        w2 = w * w  # w²
+        wabs2 = w * wcc  # ww̄
+        wcc2 = wcc * wcc  # w̄²
+        w2wcc = w * wabs2  # w²w̄
+        wwcc2 = wcc * wabs2  # ww̄²
+        w3wcc2 = wabs2 * w2wcc  # w³w̄²
+        w2wcc3 = wabs2 * wwcc2  # w²w̄³
+        M = numpy.column_stack((numpy.ones_like(w),  # zero order
+                                w, wcc,  # first order
+                                w2, wabs2, wcc2,  # second order
+                                w2wcc, wwcc2,  # third order
+                                w3wcc2, w2wcc3))  # fifth order
+        return M
+
+    def __call__(self, x):
+        x = numpy.asarray(x)
+        w = x[..., 0] + 1.0j * x[..., 1]
+
+        M = self._vandermonde(w)
+        v = numpy.dot(M, self.coeffs)
+
+        if x.ndim == 1:
+            return np.array((v.real, v.imag))
+        return numpy.column_stack((v.real, v.imag))
+
+    @classmethod
+    def from_pointset(cls, x, y):
+        """
+        Estimate the transformation from a set of corresponding points.
+
+        Constructor for AnamorphosisTransform that determines the best
+        coordinate transformation from two point sets `x` and `y` in a
+        least-squares sense. For more information see [1].
+
+        Parameters
+        ----------
+        x : (n, 2) array
+            Coordinates in the source reference frame.
+        y : (n, 2) array
+            Coordinates in the destination reference frame. Must be of same
```

Comment:
The order of the points in in x and y also need to correspond right? So if x = [point1, point5, point2, ....] then y = [point1, point5, point2, ....]

Reply:
Isn't this the case for every `from_pointset()`? If so, I'd say it's fine. Especially it's indicated as "a set of *corresponding* points".